roslibrust_codegen/src/parse.rs

```diff
@@ -331,29 +368,20 @@ fn strip_comments(line: &str) -> &str {
 fn parse_field_type(type_str: &str, is_vec: bool, pkg: &Package) -> FieldType {
     let items = type_str.split('/').collect::<Vec<&str>>();
 
-    // Select which type conversion map to use depending on package version
-    let prop_map: &HashMap<&'static str, &'static str> = match pkg.version {
-        Some(RosVersion::ROS1) => &ROS_TYPE_TO_RUST_TYPE_MAP,
-        Some(RosVersion::ROS2) => &ROS_2_TYPE_TO_RUST_TYPE_MAP,
-        None => {
-            // If we couldn't determine the package type, assume ROS1 for now
-            &ROS_TYPE_TO_RUST_TYPE_MAP
-        }
-    };
-
     if items.len() == 1 {
         // If there is only one item (no package redirect)
+        let pkg_version = pkg.version.unwrap_or(RosVersion::ROS1);
         FieldType {
-            package_name: if prop_map.contains_key(type_str) {
+            package_name: if is_intrinsic_type(pkg_version, type_str) {
                 // If it is a fundamental type, no package
-                if items[0] == "Header" {
+                None
+            } else {
+                // Otherwise it is referencing another message in the same package
+                if type_str == "Header" {
```

Comment:
Rather than treating Header as an internal type, I've changed over to explicitly assuming it comes from std_msgs here in all cases. This seems to be what some of the ROS source is doing.

Reply:
From what I've seen, the ROS tools take the same approach.